manifests/010-prometheus-rules.yaml

```diff
@@ -9,14 +9,48 @@ spec:
   groups:
   - name: SamplesOperator
     rules:
+    - alert: SamplesRetriesMissingOnImagestreamImportFailing
+      expr: sum(openshift_samples_failed_imagestream_import_info) > sum(openshift_samples_retry_imagestream_import_total) - sum(openshift_samples_retry_imagestream_import_total offset 30m)
+      for: 2h
+      labels:
+        severity: warning
+      annotations:
+        message: |
+          Samples operator is detecting problems with imagestream image imports, and the periodic retries of those
+          imports are not occurring.  Contact support.  You can look at the "openshift-samples" ClusterOperator object
+          for details. Most likely there are issues with the external image registry hosting the images that need to
+          be investigated.  The list of ImageStreams that have failing imports are:
+          {{ range query "openshift_samples_retry_imagestream_import_total > 0" }}
+            {{ .Labels.name }} {{ .Value }}
+          {{ end }}
+          However, the list of ImageStreams for which samples operator is retrying imports is:
+          retrying imports:
+          {{ range query "openshift_samples_retry_imagestream_import_total > 0" }}
+             {{ .Labels.imagestreamname }} {{ .Value }}
+          {{ end }}
```

Comment:
I don't think this will work as this might not be read as a template. Have you tested it?

Reply:
The alert fired fine when I looked at it from the prometheus UI, but I have yet tried to look at it from the alert manager UI.  I'm going to try and do that today and report back.  I'm assuming if I don't see the expected var substitution there that would confirm your suspicion.

Thanks for the quick feedback @paulfantom

Reply:
Also, one quick ask for clarification.  Looking at https://prometheus.io/docs/prometheus/latest/configuration/template_examples/#simple-alert-field-templates is the difference in that example and this that the example in the doc deals with the description, where as I'm trying to use templates for the message @paulfantom ?

Reply:
It looks like it works @paulfantom .... from the alertmanager view from the console:

```
Samples operator is detecting problems with imagestream image imports. You can look at the "openshift-samples" ClusterOperator object for details. Most likely there are issues with the external image registry hosting the images that needs to be investigated. Or you can consider marking samples opertaor Removed if you do not care about having sample imagestreams available. The list of ImageStreams for which samples operator is retrying imports: apicurito-ui 4 dotnet 4 eap-cd-openshift 17 fuse7-console 6 fuse7-karaf-openshift 12 httpd 1 java 2 jboss-amq-62 14 jboss-amq-63 6 jboss-datagrid65-client-openshift 2 jboss-datagrid65-openshift 5 jboss-datagrid71-client-openshift 3 jboss-datagrid72-openshift 6 jboss-datagrid73-openshift 5 jboss-datavirt64-openshift 8 jboss-decisionserver64-openshift 7 jboss-eap71-openshift 5 jboss-eap72-openshift 4 jboss-fuse70-eap-openshift 2 jboss-processserver64-openshift 7 jboss-webserver30-tomcat8-openshift 5 nginx 3 nodejs 3 openjdk-11-rhel7 4 openjdk-11-rhel8 3 openjdk-8-rhel8 4 perl 1 php 2 redhat-sso70-openshift 2 redhat-sso71-openshift 5 redhat-sso72-openshift 10 rhpam-businesscentral-rhel8 3 apicast-gateway 18 dotnet-runtime 2 eap-cd-runtime-openshift 2 fis-java-openshift 2 fis-karaf-openshift 4 fuse-apicurito-generator 4 fuse7-eap-openshift 6 fuse7-java-openshift 6 golang 2 jboss-datagrid71-openshift 4 jboss-datavirt64-driver-openshift 8 jboss-eap64-openshift 10 jboss-eap70-openshift 5 jboss-fuse70-console 1 jboss-fuse70-java-openshift 1 jboss-fuse70-karaf-openshift 3 jboss-webserver30-tomcat7-openshift 6 jboss-webserver31-tomcat7-openshift 5 jboss-webserver31-tomcat8-openshift 5 jboss-webserver50-tomcat9-openshift 8 mariadb 2 mongodb 5 mysql 4 postgresql 3 python 4 redhat-openjdk18-openshift 9 redhat-sso73-openshift 2 redis 2 rhdm-decisioncentral-rhel8 1 rhdm-kieserver-rhel8 1 rhpam-businesscentral-monitoring-rhel8 2 rhpam-kieserver-rhel8 2 rhpam-smartrouter-rhel8 1 ruby 3 sso74-openshift-rhel8
```

@adambkaplan FYI ^^

Reply:
Though I'm not sure where those seemingly random numbers come from ... each imagestream name is followed by a number i.e. 

`httpd 1 java 2`

investigating

Reply:
figured it out .... will push update momentarily ... new output 

Samples operator is detecting problems with imagestream image imports. You can look at the "openshift-samples" ClusterOperator object for details. Most likely there are issues with the external image registry hosting the images that needs to be investigated. Or you can consider marking samples opertaor Removed if you do not care about having sample imagestreams available. The list of ImageStreams for which samples operator is retrying imports: apicurito-ui dotnet eap-cd-openshift fuse7-console fuse7-karaf-openshift httpd java jboss-amq-62 jboss-amq-63 jboss-datagrid65-client-openshift jboss-datagrid65-openshift jboss-datagrid71-client-openshift jboss-datagrid72-openshift jboss-datagrid73-openshift jboss-datavirt64-openshift jboss-decisionserver64-openshift jboss-eap71-openshift jboss-eap72-openshift jboss-fuse70-eap-openshift jboss-processserver64-openshift jboss-webserver30-tomcat8-openshift nginx nodejs openjdk-11-rhel7 openjdk-11-rhel8 openjdk-8-rhel8 perl php redhat-sso70-openshift redhat-sso71-openshift redhat-sso72-openshift rhpam-businesscentral-rhel8 apicast-gateway dotnet-runtime eap-cd-runtime-openshift fis-java-openshift fis-karaf-openshift fuse-apicurito-generator fuse7-eap-openshift fuse7-java-openshift golang jboss-datagrid71-openshift jboss-datavirt64-driver-openshift jboss-eap64-openshift jboss-eap70-openshift jboss-fuse70-console jboss-fuse70-java-openshift jboss-fuse70-karaf-openshift jboss-webserver30-tomcat7-openshift jboss-webserver31-tomcat7-openshift jboss-webserver31-tomcat8-openshift jboss-webserver50-tomcat9-openshift mariadb mongodb mysql postgresql python redhat-openjdk18-openshift redhat-sso73-openshift redis rhdm-decisioncentral-rhel8 rhdm-kieserver-rhel8 rhpam-businesscentral-monitoring-rhel8 rhpam-kieserver-rhel8 rhpam-smartrouter-rhel8 ruby sso74-openshift-rhel8